pkg/mount/mountinfo_linux.go

```diff
@@ -114,6 +111,10 @@ func parseInfoFile(r io.Reader, filter FilterFunc) ([]*Info, error) {
 			break
 		}
 	}
+	if err = s.Err(); err != nil {
+		return nil, err
+	}
+
 	return out, nil
```

Comment:
Could probably even be simplified to (unless code is specifically expecting `nil` in the error case);

```suggestion
	return out, s.Err()
```

Reply:
Actually, let me remove this commit in favor of https://github.com/moby/moby/pull/40694